keyboards/skyloong/gk87/ptir/ptir.c

```diff
@@ -0,0 +1,264 @@
+    {1, CS1_SW5,   CS2_SW5,   CS3_SW5},
+    {1, CS1_SW6,   CS2_SW6,   CS3_SW6},
+    {1, CS1_SW7,   CS2_SW7,   CS3_SW7},
+    {1, CS1_SW8,   CS2_SW8,   CS3_SW8},
+    {1, CS1_SW9,   CS2_SW9,   CS3_SW9},
+    {1, CS1_SW10,  CS2_SW10,  CS3_SW10},
+    {1, CS1_SW11,  CS2_SW11,  CS3_SW11},
+    {1, CS10_SW1,  CS11_SW1,  CS12_SW1},
+    {1, CS10_SW2,  CS11_SW2,  CS12_SW2},
+    {1, CS10_SW3,  CS11_SW3,  CS12_SW3},
+
+    {1, CS4_SW1,   CS5_SW1,   CS6_SW1},
+    {1, CS7_SW4,   CS8_SW4,   CS9_SW4},
+    {1, CS4_SW2,   CS5_SW2,   CS6_SW2},
+    {1, CS4_SW3,   CS5_SW3,   CS6_SW3},
+    {1, CS4_SW4,   CS5_SW4,   CS6_SW4},
+    {1, CS4_SW5,   CS5_SW5,   CS6_SW5},
+    {1, CS4_SW6,   CS5_SW6,   CS6_SW6},
+    {1, CS4_SW7,   CS5_SW7,   CS6_SW7},
+    {1, CS4_SW8,   CS5_SW8,   CS6_SW8},
+    {1, CS4_SW9,   CS5_SW9,   CS6_SW9},
+    {1, CS4_SW10,  CS5_SW10,  CS6_SW10},
+    {1, CS4_SW11,  CS5_SW11,  CS6_SW11},
+    {1, CS13_SW2,  CS14_SW2,  CS15_SW2},
+    {1, CS13_SW4,  CS14_SW4,  CS15_SW4},
+
+
+    {1, CS7_SW1,   CS8_SW1,   CS9_SW1},
+    {1, CS7_SW2,   CS8_SW2,   CS9_SW2},
+    {1, CS7_SW3,   CS8_SW3,   CS9_SW3},
+    {1, CS7_SW5,   CS8_SW5,   CS9_SW5},
+    {1, CS7_SW6,   CS8_SW6,   CS9_SW6},
+    {1, CS7_SW8,   CS8_SW8,   CS9_SW8},
+    {1, CS7_SW10,  CS8_SW10,  CS9_SW10},
+    {1, CS7_SW11,  CS8_SW11,  CS9_SW11},
+    {1, CS16_SW1,  CS17_SW1,  CS18_SW1},
+    {1, CS16_SW2,  CS17_SW2,  CS18_SW2},
+    {1, CS16_SW3,  CS17_SW3,  CS18_SW3},
+    {1, CS16_SW4,  CS17_SW4,  CS18_SW4},
+    {1, CS16_SW5,  CS17_SW5,  CS18_SW5}
+};
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+
+    switch (keycode) {
+#    ifdef RGB_MATRIX_ENABLE
+        case RGB_TOG:
+            if (record->event.pressed) {
+                switch (rgb_matrix_get_flags()) {
+                    case LED_FLAG_ALL: {
+                        rgb_matrix_set_flags(LED_FLAG_NONE);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    } break;
+                    default: {
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                    } break;
+                }
+            }
+         return false;
+#    endif
+     case TO(0):
+      if (record->event.pressed) {
+       L_WIN = 1;
+      } else {
+       L_WIN = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case MO(2):
+      if (record->event.pressed) {
+       FN_WIN = 1;
+      } else {
+       FN_WIN = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case TO(1):
+      if (record->event.pressed) {
+       L_MAC = 1;
+      } else {
+       L_MAC = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case MO(3):
+      if (record->event.pressed) {
+       FN_MAC = 1;
+      } else {
+       FN_MAC = 0;
+      }
+      return true; // continue all further processing of this key
```

Comment:
```suggestion
```

This isn't a great way to handle this, because if the user uses any other keycodes (such as LT, TT, TG), then this simply won't work.

Reply:
Flag special keycodes related to layer switching to give different indicators depending on the current state of the layer. In order to realize that the layer indicator only indicates during the key operation, the key operation is completed and the exit indication is finished.